### PR TITLE
Linkedin Company Integration

### PR DESCRIPTION
**Pull request by manudev-1**
# Description of the PR

This pull request integrates **LinkedIn Company** as a possibility for the user. The integration allows the project to create prompt based on LinkedIn Company giving the name.

## Changes Made

- Added 4 public function in `LinkedinService` to manage company data
- Added 4 private function in `LinkedinService` to manage and parse company data

### CODE REVIEW

File: SocialService/Social/LinkedinService.py [hunk 1]
```diff
@@ -108,11 +105,102 @@ async def user_list(self, ai: str):
...
+            updates_list.append(updates)
+
+        Ai = AIRender(ai).model
+
+        for i, (profile, update) in enumerate(zip(profile_list, updates_list)):
+            print(f"{companies[i].title()}:")
+            print(Ai.generate_prompt_linkedin(profile, update, output, payload, ai, template, api_key=api_key))
+
+    async def single_company(self, ai: str):
+        print("\n\nPlease enter the user ID of the user you would like to scrape:")
```
Comment (alvwin) [inline SocialService/Social/LinkedinService.py]: Gotta change this to be correct for the company

File: SocialService/Social/LinkedinService.py [hunk 2]
```diff
@@ -108,11 +105,102 @@ async def user_list(self, ai: str):
...
+        generation_option = self._linkedin_generation_options()
+        payload, payload_text = self._payload_options()
+        template = self._template_options()
+
+        Ai = AIRender(ai).model
+
+        print(Ai.generate_prompt_linkedin(company, updates, generation_option, payload, ai, template, payload_text, api_key=""))
+
+    async def company_list(self, ai: str):
+        print("\n\nPlease enter the path to the file containing the list of usernames you would like to scrape:")
```
Comment (alvwin) [inline SocialService/Social/LinkedinService.py]: here as well `Please enter the path to the file containing the list of companies you would like to scrape:`

File: SocialService/Social/LinkedinService.py [hunk 3]
```diff
@@ -108,11 +105,102 @@ async def user_list(self, ai: str):
...
+        for user in company:
+            if '\n' in user:
+                user = user[:-1]
+            clean_usernames.append(user)
+        profile_list = []
+        posts_list = []
+        for user_id in clean_usernames:
+            company, updates = self._get_company(user_id)
+            if company == None:
+                print(f"{Const.COLOR_ERROR}User {user_id} not found{Const.RESET_ALL}")
```
Comment (alvwin) [inline SocialService/Social/LinkedinService.py]: And here

**Review by alvwin — CHANGES_REQUESTED** [review]
Needs a few changes in console outputs and i haven't had the time to test it yet

**Comment by manudev-1** [comment]
> Needs a few changes in console outputs and i haven't had the time to test it yet

I will apply the changes that you suggest

**Review by alvwin — COMMENTED** [review]
Still need to test but code looks good

**Review by alvwin — APPROVED** [review]

**Comment by manudev-1** [comment]
> Still need to test but code looks good

Yeah, sure no problem, take your time

File: Util/Const.py [hunk 4]
```diff
@@ -15,4 +16,5 @@ class Const:
     generation_options_list_twitter = ["Email", "SMS", "Vishing script", "Twitter DM", "Twitter post", "Twitter reply", "Pretext"]
     generation_options_list_linkedin = ["Email", "SMS", "Vishing script", "Pretext"]
     payload_options_list = ["Login page", "Attachment", "Download Link", "Else"]
-    user_options_list = ["Single user", "User list"]
\ No newline at end of file
+    user_options_list = ["Single user", "User list"]
+    account_type_options_list = ["User", "Company"]
```
Comment (alvwin) [inline Util/Const.py:21]: needs an company_options_list "single company" and "multiple companies"

File: console_program.py [hunk 5]
```diff
@@ -17,37 +16,52 @@ async def main(args=None):
...
       ai_option = Const.ai_options_list[ai_options()-1]
 
       platform_option = platform_options()
+      if platform_option == 2:
+            account_type = account_type_options()
       mult_option = mult_options()
 
       social = SocialRender(Const.platform_options_list[platform_option - 1]).service
 
+      if account_type == 2:
```
Comment (alvwin) [inline console_program.py:64]: throws an exception when platform_option == 1

File: SocialService/Social/LinkedinService.py [hunk 6]
```diff
@@ -108,11 +105,102 @@ async def user_list(self, ai: str):
...
+            print(profile.split("\n")[0])
+            print(Ai.generate_prompt_linkedin(profile, update, output, payload, ai, template, api_key=api_key))
+
+    async def single_company(self, ai: str):
+        print("\n\nPlease enter the company ID of the company you would like to scrape:")
+        print("(the company ID is in the URL of the company's profile like this: https://www.linkedin.com/company/company-id/)")
+        company_id = input("\nCompany ID: ")
+        company, updates = self._get_company(company_id)
+        if company is None:
+            print(f"{Const.COLOR_ERROR}Company not found{Const.RESET_ALL}")
```
Comment (alvwin) [inline SocialService/Social/LinkedinService.py:156]: The error when the company doesn't exist doesn't work this way, better with try/catch

**Review by alvwin — CHANGES_REQUESTED** [review]

**Review by alvwin — APPROVED** [review]

**Pull request merged**